dapps/src/api/time.rs

```diff
@@ -56,27 +64,72 @@ impl From<io::Error> for Error {
 	fn from(err: io::Error) -> Self { Error::Io(format!("{}", err)) }
 }
 
-impl From<fetch::Error> for Error {
-	fn from(err: fetch::Error) -> Self { Error::Fetch(format!("{}", err)) }
+impl From<ntp::errors::Error> for Error {
+	fn from(err: ntp::errors::Error) -> Self { Error::Ntp(format!("{}", err)) }
 }
 
-
 /// Time provider.
 pub trait TimeProvider: Clone + Send + 'static {
 	/// Returns an instance of this provider.
 	fn new() -> Self where Self: Sized;
-	/// Returns current time in milliseconds.
-	fn utc_timestamp_millis(&self) -> i64;
+	/// Returns current time.
+	fn now(&self) -> Timespec;
 }
 /// Default system time provider.
 #[derive(Clone)]
 pub struct StdTimeProvider;
 impl TimeProvider for StdTimeProvider {
 	fn new() -> Self where Self: Sized { StdTimeProvider }
-	fn utc_timestamp_millis(&self) -> i64 {
-		let time = ::time::now_utc().to_timespec();
+	fn now(&self) -> Timespec {
```

Comment:
We can inline this function since we  don't really test internals of `SimpleNtp`.